### DISCUSSION TITLE
Try new Markdown SQL Pro Tools with DuckDB Pro extension for VS Code IDE today

### DISCUSSION THREAD

**Discussion by RandomFractals** [mtxr/vscode-sqltools#1199] [Show and tell]
📢 We released new version of [Markdown SQL Pro Tools](https://randomfractals.github.io/pro-data-tools/#markdown-sql-pro-tools) for VS Code IDE that integrates with all the SQL Tools drivers, including our free [DuckDB Sql Tools](https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.duckdb-sql-tools) and premium [DuckDB Pro Tools](https://randomfractals.github.io/pro-data-tools/#duckdb-pro-tools).

New **SQL Code Lenses**, features, and walk-throughs include:

✅ SQL code block highlights and selections in markdown documents

✅ Detection of `%%sql` magic code blocks and `%sql` magic code lines with SELECT statements in various markdown documents

✅ New [SQLBookmarks](https://twitter.com/hashtag/SQLBookmarks?src=hashtag_click) 📑 for markdown documents with SQL code blocks

✅ `Extract All` SQL code from markdown code fences to create new SQL document

✅ MyST [JupyterBook](https://twitter.com/search?q=%23JupyterBook&src=typed_query&f=top) markdown documents support with SQL code lenses in code cells with SQL magics

✅ Interactive SQL support and code lenses for [Quarto](https://quarto.org/) docs with SQL code blocks

✅ Interactive SQL support and code lenses for SQL code chunks in [Rmarkdown](https://rmarkdown.rstudio.com/) documents

✅ New SQL code cells support and SQL code lenses  for [ObservableJS](https://marketplace.visualstudio.com/items?itemName=GordonSmith.observable-js) and markdown documents

**Markdown SQL Pro** extension documentation and new demos at:

📰 ⮚ https://randomfractals.github.io/pro-data-tools/#markdown-sql-pro-tools

This SQL Tools extension and our **DuckDB Pro Tools** for VS Code IDE are free today, for one day only. Sign up and try them out!

💖 ⮚  https://github.com/sponsors/RandomFractals/sponsorships?tier_id=295482

Related prior post covering advanced **DuckDB Pro Tools** features: https://github.com/duckdb/duckdb/discussions/7388

![markdown-sql-pro-tools](https://github.com/mtxr/vscode-sqltools/assets/656833/743df5ab-9527-486f-8c5d-d8e230e53cc5)

![duckdb-tools-extensions](https://github.com/mtxr/vscode-sqltools/assets/656833/b84f2734-a49a-4fb3-8486-1a6c71090352)

**Comment by gjsjohnmurray** [comment]
Impressive work!

> free today, for one day only

Can I be the first to ask how we access this offer (today being 4th August 2023)?

**Reply by RandomFractals** [reply]
Subscriptions are manual for now. Once you sign up for one of the **Pro** sponsor tiers on Github, we'll add you as collaborator to grant you access to the individual Pro Data Tools extension docs and `.vsix` package to download from release tags and install in VS Code IDE.

We plan to automate that process this month, publish them in marketplace, and provide a limited set of features free, with some of the advanced Pro features that will be available only to our Pro sponsors on GitHub, similar to how GitLens extension does it.

Our Github Sponsorship page to sign up for now: https://github.com/sponsors/RandomFractals

**Comment by RandomFractals** [comment]
We put together new documentation page covering our DuckDB VS Code SQL Tools extensions:

https://randomfractals.github.io/pro-data-tools/duckdb-tools

There is also a new GitHub Discussions portal for the VS Code users interested in our new SQL-based Pro Data Tools:

https://github.com/RandomFractals/pro-data-tools/discussions
